### PR TITLE
plugins: introduce v2.RustPlugin

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/plugins/v2/rust.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        return {"curl", "gcc", "git"}
+
+    def get_build_environment(self) -> Dict[str, str]:
+        return {"PATH": "${HOME}/.cargo/bin:${PATH}"}
+
+    def _get_rustup_command(self) -> str:
+        return dedent(
+            """\
+        if ! command -v rustup 2>/dev/null; then
+            curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --profile=minimal
```
Comment (cjp256) [inline snapcraft/plugins/v2/rust.py:98]: I'm curious what the rationale is for the proto and tls flags? :)

Comment (sergiusens) [reply snapcraft/plugins/v2/rust.py:98]: That's the recommendation upstream https://rustup.rs/

Comment (cjp256) [reply snapcraft/plugins/v2/rust.py:98]: It is indeed, interesting.

File: tests/spread/plugins/v1/rust/cargo-lock/task.yaml [hunk 2]
```diff
@@ -1,47 +0,0 @@
-summary: Build and run a basic rust snap with Cargo.lock
```
Comment (cjp256) [inline tests/spread/plugins/v1/rust/cargo-lock/task.yaml:1]: Deleted?

Comment (sergiusens) [reply tests/spread/plugins/v1/rust/cargo-lock/task.yaml:1]: yes, we implicitly run these tests with the other tests that already have a lock.

Comment (sergiusens) [reply tests/spread/plugins/v1/rust/cargo-lock/task.yaml:1]: Agreed to add this back for v1

**Pull request merged**